### PR TITLE
Support for blocking commands

### PR DESCRIPTION
**Pull request by Sasidharan-Gopal**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/clusters/redis/redis_cluster.cc [hunk 1]
```diff
@@ -297,7 +297,7 @@ void RedisCluster::RedisDiscoverySession::startResolveRedis() {
     client->host_ = current_host_address_;
     client->client_ = client_factory_.create(host, dispatcher_, *this, redis_command_stats_,
                                              parent_.info()->statsScope(), parent_.auth_username_,
-                                             parent_.auth_password_, false,false,nullptr);
+                                             parent_.auth_password_, false,false,false,nullptr);
```
Comment (dinesh-murugiah) [inline source/extensions/clusters/redis/redis_cluster.cc:300]: we can try to optimise this by accepting an enum as input, and setting booleans appropriately

File: source/extensions/filters/network/common/redis/client_impl.cc [hunk 2]
```diff
@@ -167,7 +169,9 @@ void ClientImpl::onConnectOrOpTimeout() {
     host_->stats().cx_connect_fail_.inc();
   }
 
-  connection_->close(Network::ConnectionCloseType::NoFlush);
+  if (!is_blocking_client_) {
```
Comment (dinesh-murugiah) [inline source/extensions/filters/network/common/redis/client_impl.cc:172]: we can have a debug level log here, just to indicate that the blocking client is ignoring optimeout

File: source/extensions/filters/network/redis_proxy/proxy_filter.cc [hunk 3]
```diff
@@ -192,11 +192,13 @@ void ProxyFilter::onAsyncResponse(Common::Redis::RespValuePtr&& value){
...
 
   // Check if there is an active transaction that needs to be closed.
-  if (transaction_.should_close_ ) {
+  if (transaction_.should_close_ || transaction_.is_blocking_command_) {
       //Close all upsteam clients and ref to pubsub callbacks if any
       transaction_.close();
       // decrement the reference to proxy filter
-      transaction_.setPubsubCallback(nullptr);
+      if (!transaction_.is_blocking_command_ ) {
+        transaction_.setPubsubCallback(nullptr);
```
Comment (dinesh-murugiah) [inline source/extensions/filters/network/redis_proxy/proxy_filter.cc:200]: This has to set irrespective of any command type , i wil fix it as part of my fixes

Comment (Sasidharan-Gopal) [reply source/extensions/filters/network/redis_proxy/proxy_filter.cc:200]: Okay

File: source/extensions/filters/network/common/redis/client_impl.cc (hunk 2 above)
Comment (Sasidharan-Gopal) [reply source/extensions/filters/network/common/redis/client_impl.cc:172]: Sure, will add

File: source/extensions/clusters/redis/redis_cluster.cc (hunk 1 above)
Comment (Sasidharan-Gopal) [reply source/extensions/clusters/redis/redis_cluster.cc:300]: Ok ji

**Pull request merged**